testConnector/src/test/java/io/vantiq/extsrc/testConnector/TestTestConnectorCore.java

```diff
@@ -127,6 +126,30 @@ public void testProcessRequest() {
         assert resultEnvVars.get(NONEXISTENT_ENV_VAR) == null;
     }
 
+    @Test
+    public void testRawBytes() {
+        assumeTrue(filename != null && fileContents != null);
+
+        // Put data in the request and call the process method
+        Map<String, Object> request = new LinkedHashMap<>();
+        List<String> filenames = new ArrayList<>();
+        filenames.add(filename);
+        request.put("filenames", filenames);
+        request.put("rawBytes", true);
+        Map result = core.processRequest(request, null);
+
+        // Check that the data is base64 encoded version of what we expect
+        assert result.get("files") instanceof Map;
+        Map resultFiles = (Map) result.get("files");
+        assert resultFiles.get(filename) instanceof String;
+        String fileString = (String) resultFiles.get(filename);
+        assert !fileString.isEmpty();
+        assert !fileString.equals(fileContents);
+        String decodedString = new String(Base64.getDecoder().decode(fileString.getBytes()));
+        decodedString = decodedString.replace(System.getProperty("line.separator"), ""); // Removing new lines
```

Comment:
Why are there any newlines?  if we put binary data in, there should be no newlines (unless they were in the binary data) on output.  The base64 encoded data may have newlines or other whitespace, but the decoding should remove them.  So I don't know why there's a removal of newlines after decoding.

Maybe this is because of the stringificiation above?

Reply:
I couldn't quite figure out where things were going wrong.

I've created a test file on my machine that I used for this test. It just contains a few words. Initially this test kept on failing because somewhere along the road a `\n` is added to the end of the base64 encoded string. I figured it must be something related to the Mac OS adding a new line to the end of files, so for simplicities sake I just chose to remove that new line.